tests/test_freqtradebot.py

```diff
@@ -1533,37 +1558,36 @@ def test_tsl_on_exchange_compatible_with_edge(mocker, edge_conf, fee, caplog,
 
     # price goes down 5%
     mocker.patch('freqtrade.exchange.Exchange.fetch_ticker', MagicMock(return_value={
-        'bid': 0.00001172 * 0.95,
-        'ask': 0.00001173 * 0.95,
-        'last': 0.00001172 * 0.95
+        'bid': 1.9 * 0.95,
+        'ask': 2.2 * 0.95,
+        'last': 1.9 * 0.95
     }))
-
     assert freqtrade.handle_trade(trade) is False
```

Comment:
This test is failing with 

```
>       assert freqtrade.handle_trade(trade) is False

tests/test_freqtradebot.py:1565: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
freqtrade/freqtradebot.py:717: in handle_trade
    if self._check_and_execute_exit(trade, exit_rate, buy, sell):
freqtrade/freqtradebot.py:866: in _check_and_execute_exit
    self.execute_trade_exit(trade, exit_rate, should_sell)
freqtrade/freqtradebot.py:1126: in execute_trade_exit
    order = self.exchange.create_order(pair=trade.pair,
/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/mock.py:1093: in __call__
    return self._mock_call(*args, **kwargs)
/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/mock.py:1097: in _mock_call
    return self._execute_mock_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock id='4940277168'>, args = ()
kwargs = {'amount': 13.15657894, 'ordertype': 'limit', 'pair': 'NEO/BTC', 'rate': 2.178, ...}
effect = <list_iterator object at 0x1269a07f0>

    def _execute_mock_call(self, /, *args, **kwargs):
        # separate from _increment_mock_call so that awaited functions are
        # executed separately from their call, also AsyncMock overrides this method
    
        effect = self.side_effect
        if effect is not None:
            if _is_exception(effect):
                raise effect
            elif not _callable(effect):
>               result = next(effect)
E               StopIteration

/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/mock.py:1154: StopIteration
```

Not really sure what this means or how to fix it

Reply:
Your mock was exhausted (which can only happen when using a list as part of `side_effect` to the mock).
probably, now there's more trades (for whatever reason) - maybe you've now got more money left (different stake_amount/wallet ratio), so more trades can be filled...